src/test/java/core/basesyntax/service/impl/ParserImplTest.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ParserImplTest {
+    private ParserImpl parser;
+
+    @BeforeEach
+    void setUp() {
+        parser = new ParserImpl();
+    }
+
+    @Test
+    void parseValidRecords_ok() {
+        List<String> records = Arrays.asList("b,apple,10",
+                "s,banana,20", "p,orange,30", "r,pear,40");
+        List<FruitTransaction> transactions = parser.parse(records);
+        assertEquals(4, transactions.size(),
+                "Should parse all records correctly.");
+    }
+
+    @Test
+    void parseInvalidFormat_notOk() {
+        List<String> records = List.of("b,apple");
```

Comment:
You can make this List a class field and initialize it in each method and after test use record.clear